### ISSUE TITLE
Did we already fix https://github.com/lowRISC/ibex/issues/1719?

### ISSUE THREAD

**Issue by Silabs-ArjanB** [openhwgroup/cv32e40s#275]
See https://github.com/lowRISC/ibex/issues/1719

**Comment by silabs-robin** [comment]
I have made a cover property showing that we can exhibit correct behavior for the case in that issue.
(I left a todo to make an assert for this once we have reliable rvfi csr wdata signals.)

The original issue is confusing and I can't see clearly what they actually claim.
I tried to decipher their log, and it seems Spike is in the wrong and the Ibex rtl is right.
This is also confirmed by their comments further down, linking to a PR to Spike.

Ibex never had a problem, if we borrowed code properly then we never had a problem, and my cover indicates that we indeed do not have a problem?

![image](https://user-images.githubusercontent.com/75431602/183632151-fa6bf94f-1d0a-4086-8914-3926d1ff438e.png)

```
setup-state:
5843330f

setup-equivalent:
1803130f


initial-cfg:
17978d0c

initial-locks:
00111100

initial-tor:
00001111


write-attempt:
5843330f

write-equivalent:
1803130f

write-locks:
00000000


result-cfg:
18978d0f

result-changed:
11000011  // Cfg0 changed as expected


spike-cfg:
18978d0c

spike-changed:
11000000  // Cfg0 refused to change
```
